### PR TITLE
clamav: 0.103.5 -> 0.105.0

### PR DESCRIPTION
**Pull request by lheckemann**
###### Description of changes

This goes beyond #172835 to update clamav past the [build system change](https://github.com/Cisco-Talos/clamav/blob/rel/0.105/NEWS.md#new-requirements). Benefits include a smaller expression and the full manual being installed.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rapenne-s** [comment]
Result of `nixpkgs-review pr 172851` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clamav</li>
  </ul>
</details>

File: pkgs/tools/security/clamav/default.nix [hunk 1]
```diff
@@ -1,52 +1,34 @@
...
-    mkdir $out/etc
-    cp etc/*.sample $out/etc
-  '';
+  cmakeFlags = [
+    "-DSYSTEMD_UNIT_DIR=${placeholder "out"}/lib/systemd"
+  ];
 
   # Only required for the unit tests
-  hardeningDisable = [ "format" ];
+  #hardeningDisable = [ "format" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/clamav/default.nix:31]: The one that is disabled or all and is this still up to date?

Comment (lheckemann) [reply pkgs/tools/security/clamav/default.nix:31]: Whoops, I tried just commenting it out to see if it's still necessary and then forgot to remove it completely after seeing that the tests now work without disabling that hardening.

**Review by ajs124 — APPROVED** [review]

**Comment by KoviRobi** [comment]
Between this and 9e1e3b288062abf37847307adeee17f3c8dddc42 I get a problem of
```
May 18 11:56:05 pc-nixos-a systemd[1]: Starting ClamAV daemon (clamd)...
May 18 11:56:05 pc-nixos-a systemd[1]: Started ClamAV daemon (clamd).
May 18 11:56:05 pc-nixos-a clamd[147557]: ERROR: Can't open/parse the config file /nix/store/q8ppba15a7qpxkm0xv1pbv86pyriw6w3-clamav-0.105.0/etc/clamd.conf
May 18 11:56:05 pc-nixos-a systemd[1]: clamav-daemon.service: Main process exited, code=exited, status=1/FAILURE
May 18 11:56:05 pc-nixos-a systemd[1]: clamav-daemon.service: Failed with result 'exit-code'.
```

Given 9e1e3b288062abf37847307adeee17f3c8dddc42 has in the commit log:

> Fixes console tools to work as expected as they require hardcoded config
> file locations.

I wonder if it is still appropriate, and we need `sysconfdir`, or alternatively we need `--config-file=${clamdConfigFile}` (and likewise for freshclam) in the NixOS service files.

**Pull request merged**